content/blog/2022-05-17-aws-remotes-in-dvc.md

```diff
@@ -0,0 +1,138 @@
+see the success message and your new bucket.
+
+![newly created S3 bucket in AWS](/uploads/images/2022-05-17/created_bucket.png)
+
+### Get your credentials
+
+Now that the S3 bucket is ready, we need the `access_key_id` and
+`secret_access_key` from AWS in order to connect to our project. You can create
+these keys in your Identity and Access Management settings. Go to your security
+credentials and select the `Access keys` section. Then click the
+`Create New Access Key` button. This will generate a new set of keys for you so
+make sure you download this file to get your secret access key.
+
+![make AWS access credentials](/uploads/images/2022-05-17/make_credentials.png)
+
+Once you’ve downloaded the credentials, you should see the access key ID in the
+table. Note that you won’t be able to access your secret key again at this
+point. You would need to make a new set of credentials if you don’t have it.
+
+![successfully created AWS access credentials](/uploads/images/2022-05-17/credentials.png)
+
+That’s it for setting up your bucket and getting the credentials you need! Now
+let’s add DVC to our demo repo and set up the remote.
+
+## Set up a DVC project
+
+First, add DVC as a requirement to your project with the following installation
+command:
+
+`$ pip install 'dvc[s3]'`
+
+Then you can initialize DVC in your own project with the following command:
+
+`$ dvc init`
+
+This will add all of the DVC internals needed to start versioning your data and
+tracking experiments. Now we need to set up the remote to connect our project
+data stored in AWS to the DVC repo.
+
+### Create a default remote
+
+Now we can add a default to the project with the following command:
+
+`$ dvc remote add -d bikes s3://updatedbikedata/train/`
+
+This creates a default remote called `bikes` that connects to the
+`updatedbikedata` bucket we made earlier and it references a folder inside this
+bucket called `train` which is where the training data for the model will be
+stored.
+
+### Add AWS credentials
+
+In order for DVC to be able to push and pull data from the remote, you need to
+have valid AWS credentials.
+
+If you are using the AWS CLI already, it’s likely that your credentials are
+available through the profiles. You can check that out
+[here in the AWS docs](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html).
+
+If you run into issues with the credentials, you can manually add them with the
+following commands:
+
+`$ dvc remote modify --local bikes access_key_id 'mykey'`
+
+`$ dvc remote modify --local bikes secret_access_key 'mysecret'`
+
+### Push and pull data with DVC
+
+Now you can push data from your local machine to the AWS remote! First, add the
+data you want DVC to track with the following command:
+
+`$ dvc add data`
+
+This will allow DVC to track the entire `data` directory so it will note when
+any changes are made. Then you can push that data to your AWS remote with this
+command:
+
+`$ dvc push`
```

Comment:
How DVC stores the data in the remote is a common question. Maybe we could show how the uploaded data looks in the AWS UI, and potentially add a link to https://dvc.org/doc/command-reference/push#example-what-happens-in-the-cache .

Reply:
Other relevant content related to this: https://dvc.org/doc/user-guide/project-structure/internal-files#structure-of-the-cache-directory